monster.cpp

```diff
@@ -1,30 +1,63 @@
 #include "monster.hpp"
+#include <fstream>
+#include <iostream>
 
-std::string Monster::getName() const {
-  return name;
-}
+std::string Monster::getName() const { return name; }
 
-int Monster::getHp() const {
-  return hp;
-}
+int Monster::getHp() const { return hp; }
 
-int Monster::getDmg() const {
-  return dmg;
-}
+int Monster::getDmg() const { return dmg; }
 
-void Monster::hpDown(int dmg) {
-  hp -= dmg;
-}
+void Monster::hpDown(int dmg) { hp -= dmg; }
 
 void Monster::monsterAttack(Monster &monster) {
-  if (hp > 0) {
-    monster.hpDown(dmg);
-  }
+  if (hp > 0) monster.hpDown(dmg);
 }
 
 bool Monster::isDefeated() const {
-  if (hp <= 0) {
-    return true;
-  }
+  if (hp <= 0) return true;
   else return false;
 }
+
+Monster Monster::readJson(const std::string &filename) {
+  std::ifstream file(filename);
+
+  if (file.is_open()) {
+
+    std::string name;
+    int hp;
+    int dmg;
+    std::string line;
+
+    std::getline(file,line);
+
+    int posColon;
+    int posComma;
+
+    std::getline(file,line);
+    posColon = line.find(':');
+    posComma = line.find(',');
+    name = line.substr(posColon+3,(posComma-1)-(posColon+3));
```

Comment:
Ennyire azért ne legyen érzékeny a whitespace-ekre. A kettőspont és a vessző körül tetszőleges számú szóköz és tabulátor lehet. (A valóságban sortörés is, de azt az esetet most még nem muszáj tudni kezelni.)
És annyit azért ellenőrizzen le, hogy tényleg az a kulcs, hogy `"name"`.

Reply:
@hmartin28 Ez a rész nem lett javítva. De mivel a json parsert úgyis tovább kell fejleszteni, egyelőre elfogadom így.